File: 1260-shift-2d-grid/1260-shift-2d-grid.cpp
class Solution {
public:
    vector<vector<int>> shiftGrid(vector<vector<int>>& grid, int k) {
        vector<int> vec;
        for(auto &it: grid){
            for(auto &it1: it) vec.push_back(it1);
        }
        k = k % vec.size();
        reverse(vec.begin(),vec.end());
        reverse(vec.begin(),vec.begin()+k);
        reverse(vec.begin()+k,vec.end());
        
        int x = 0;
        for(int i=0;i<grid.size();i++){
            for(int j=0;j<grid[0].size();j++){
                grid[i][j] = vec[x];
                x++;
            }
        }
        return grid;
    }
};

File: 1260-shift-2d-grid/1260-shift-2d-grid.py
class Solution:
    def shiftGrid(self, grid: List[List[int]], k: int) -> List[List[int]]:
        l = []
        for i in grid:
            for j in i:
                l.append(j)
                
        k = k % len(l)
        
        l = l[-k:] + l[:-k]
                
        ans = []
        x = 0
        for i in range(len(grid)):
            temp = []
            for j in range(len(grid[i])):
                temp.append(l[x])
                x+=1
            ans.append(temp)
        return ans